shell/browser/api/electron_api_session.cc

```diff
@@ -150,6 +154,42 @@ uint32_t GetQuotaMask(const std::vector<std::string>& quota_types) {
   return quota_mask;
 }
 
+constexpr content::BrowsingDataRemover::DataType kClearDataTypeAll =
+    0xffffffffffffffffull;
+constexpr content::BrowsingDataRemover::OriginType kClearOriginTypeAll =
+    content::BrowsingDataRemover::ORIGIN_TYPE_UNPROTECTED_WEB |
+    content::BrowsingDataRemover::ORIGIN_TYPE_PROTECTED_WEB;
+
+// Observes the BrowsingDataRemover that backs the `clearBrowsingData` method
+// and resolves/rejects that API's promise once its done
+class ClearBrowsingDataObserver
+    : public content::BrowsingDataRemover::Observer {
+ public:
+  ClearBrowsingDataObserver(gin_helper::Promise<void> promise,
+                            content::BrowsingDataRemover* remover)
+      : promise_(std::move(promise)) {
+    observation_.Observe(remover);
+  }
+
+  void OnBrowsingDataRemoverDone(
+      content::BrowsingDataRemover::DataType failed_data_types) override {
```

Comment:
_(style nit)_ we're inside a BrowsingDataRemover subclass, so the namespace is implicit

```suggestion
  void OnBrowsingDataRemoverDone(DataType failed_data_types) override {
```

Reply:
This makes sense, but Clang doesn't seem to be able to locate the type without the full namespaced path, so I can't make this change.

Reply:
Huh, ok. Sorry for the noise then :smile_cat: